service/server.py

```diff
@@ -114,33 +109,33 @@ def create_json(address_rec, latest_sale):
         'street': address_rec.thoroughfareName.rstrip(),
         'town': address_rec.postTown.rstrip(),
         'county': address_rec.dependentLocality.rstrip(),
-        'postcode': address_rec.postcode.rstrip(),
+        'postcode': address_rec.postCode.rstrip(),
         'amount': latest_sale.get('amount', None),
         'date': latest_sale.get('date', None),
         'property_type':
             get_property_type(latest_sale.get('property_type', None)),
         'coordinates' : {
-            'latitude': address_rec.positionY,
-            'longitude': address_rec.positionX,
+            'latitude': address_rec.position.y,
+            'longitude': address_rec.position.x,
         },
     }
-    return result
 
+    return result
 
-@app.route('/properties/<postcode>/<street_paon_saon>', methods=['GET'])
-def get_property(postcode, street_paon_saon):
-    field_vals = [postcode] + street_paon_saon.split('_')
-    check_field_vals(field_vals)
-    query_dict = get_query_dict(field_vals)
 
-    address_recs = get_property_address(query_dict)
+@app.route('/properties/<postcode>/<joined_address_fields>', methods=['GET'])
+def get_property(postcode, joined_address_fields):
+    address_key = '{}_{}'.format(joined_address_fields, postcode).upper()
+    address_recs = get_property_address(address_key)
     nof_results = len(address_recs)
     if nof_results != 1:
         abort(404)
+    address_rec = address_recs[0]
 
+    query_dict = get_query_dict(address_rec)
     latest_sale = get_latest_sale(query_dict)
 
-    result = create_json(address_recs[0], latest_sale)
```

Comment:
We have more places that need logging (500/404 error handling method, e.g.). Logging is not set up, yet. I'm working on it, but it's a separate story.